src/AutoRest.CSharp/Common/Output/Expressions/Statements/ForeachStatement.cs

```diff
@@ -9,25 +9,30 @@
 
 namespace AutoRest.CSharp.Common.Output.Expressions.Statements
 {
-    internal record ForeachStatement(CodeWriterDeclaration Item, ValueExpression Enumerable, bool IsAsync = false) : MethodBodyStatement, IEnumerable<MethodBodyStatement>
+    internal record ForeachStatement(CSharpType? ItemType, CodeWriterDeclaration Item, ValueExpression Enumerable, bool IsAsync = false, bool UseVarAsItemType = false) : MethodBodyStatement, IEnumerable<MethodBodyStatement>
```

Comment:
There is a conversion across expressions that `null` value for type property automatically means `var`. We can use it here as well.

Reply:
Could you please describe this in a bit details? I did not quite follow this

Reply:
Resolved